sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/MetricsAdvisorClient/MetricFeedbackLiveTests.cs

```diff
@@ -0,0 +1,290 @@
+            dimensionKey.AddDimensionColumn("category", "Handmade");
+
+            var filter = new FeedbackDimensionFilter(dimensionKey);
+            var comment = "Feedback created in a .NET test.";
+
+            var feedbackToAdd = new MetricCommentFeedback(MetricId, filter, comment);
+
+            if (populateOptionalMembers)
+            {
+                feedbackToAdd.StartTime = CreatedFeedbackStartTime;
+                feedbackToAdd.EndTime = CreatedFeedbackEndTime;
+            }
+
+            string feedbackId = await client.AddFeedbackAsync(feedbackToAdd);
+
+            Assert.That(feedbackId, Is.Not.Null);
+
+            MetricFeedback addedFeedback = await client.GetFeedbackAsync(feedbackId);
+
+            ValidateMetricFeedback(addedFeedback, feedbackId, dimensionKey);
+
+            Assert.That(addedFeedback.Type, Is.EqualTo(FeedbackType.Comment));
+
+            var commentFeedback = addedFeedback as MetricCommentFeedback;
+
+            Assert.That(commentFeedback, Is.Not.Null);
+            Assert.That(commentFeedback.Comment, Is.EqualTo(comment));
+
+            if (populateOptionalMembers)
+            {
+                Assert.That(commentFeedback.StartTime, Is.EqualTo(CreatedFeedbackStartTime));
+                Assert.That(commentFeedback.EndTime, Is.EqualTo(CreatedFeedbackEndTime));
+            }
+            else
+            {
+                Assert.That(commentFeedback.StartTime, Is.Null);
+                Assert.That(commentFeedback.EndTime, Is.Null);
+            }
+        }
+
+        [RecordedTest]
+        public async Task AddAndGetPeriodFeedback()
+        {
+            MetricsAdvisorClient client = GetMetricsAdvisorClient();
+
+            var dimensionKey = new DimensionKey();
+            dimensionKey.AddDimensionColumn("city", "Delhi");
+            dimensionKey.AddDimensionColumn("category", "Handmade");
+
+            var filter = new FeedbackDimensionFilter(dimensionKey);
+            var periodValue = 10;
+
+            var feedbackToAdd = new MetricPeriodFeedback(MetricId, filter, PeriodType.AutoDetect, periodValue);
+
+            string feedbackId = await client.AddFeedbackAsync(feedbackToAdd);
+
+            Assert.That(feedbackId, Is.Not.Null);
+
+            MetricFeedback addedFeedback = await client.GetFeedbackAsync(feedbackId);
+
+            ValidateMetricFeedback(addedFeedback, feedbackId, dimensionKey);
+
+            Assert.That(addedFeedback.Type, Is.EqualTo(FeedbackType.Period));
+
+            var periodFeedback = addedFeedback as MetricPeriodFeedback;
+
+            Assert.That(periodFeedback, Is.Not.Null);
+            Assert.That(periodFeedback.PeriodType, Is.EqualTo(PeriodType.AutoDetect));
+            Assert.That(periodFeedback.PeriodValue, Is.EqualTo(periodValue));
+        }
+
+        [RecordedTest]
+        [TestCase(true)]
+        [TestCase(false)]
+        public async Task GetAllFeedback(bool populateOptionalMembers)
```

Comment:
are u concerned that the test depends on existing feedback in the resource? what if the resource is empty and for whatever reason, this is the first test to run?
It will be better to always add some values and then do the getall...

Reply:
We know for sure there's feedback in the sampling time range, and feedback can't be deleted once it's added, so I think that's not a major concern.

If for any reason we need to reset our test resource, a lot of tests are expected to fail until things are in place (service has finished data ingestion, detected anomalies, triggered alerts, our hardcoded IDs have been replaced, etc). Unfortunately, the nature of this service makes it impossible for us to test things reliably from a brand new resource. We're always assuming we have a resource set up and ready to be tested in place.

Reply:
Ahh good point. ok, then all good